test/KBaseReport_server_test.py

```diff
@@ -268,6 +499,51 @@ def test_create_extended_report_with_html_paths(self):
         })
         self.check_extended_result(result, 'html_links', ['index.html', 'b'])
 
+    def test_create_extended_report_with_templates(self):
+
+        TEST_DATA = get_test_data()
+        ref_text = TEST_DATA['render_test']['content']
+#         direct_html = report_data['direct_html']
+#         self.assertEqual(direct_html.rstrip(), ref_text['abs_path'])
+
+        tmpl_arr = [
+            {
+                'name': 'none',
+                'template': {
+                    'template_file': TEST_DATA['template'],
+                },
+            },
+            {
+                'name': 'content',
+                'template': {
+                    'template_file': TEST_DATA['template'],
+                    'template_data_json': TEST_DATA['content_json'],
+                },
+            },
+            {
+                'name': 'title',
+                'template': {
+                    'template_file': TEST_DATA['template'],
+                    'template_data_json': TEST_DATA['title_json'],
+                },
+            }
+        ]
+        result = self.getImpl().create_extended_report(self.getContext(), {
+            'workspace_name': self.getWsName(),
+            'report_object_name': 'my_report',
+            'direct_html_link_index': 0,
+            'html_links': tmpl_arr,
+        })
+        self.check_extended_result(result, 'html_links', ['none', 'content', 'title'])
+
+        # use the same templates to generate files
+        result = self.getImpl().create_extended_report(self.getContext(), {
+            'workspace_name': self.getWsName(),
+            'report_object_name': 'my_report',
+            'file_links': tmpl_arr,
+        })
+        self.check_extended_result(result, 'file_links', ['none', 'content', 'title'])
```

Comment:
I was going to add in some code here to download the files from shock and check the content was as expected, but ran out of test energy. I checked manually.